sinonjs/referee-sinon
# Use new mochify, update dependencies, drop old node

#### Purpose (TL;DR) - mandatory

- Update all dependencies.
- Update GitHub actions.
- Drop node 12 and 14
- Support node 18 and 20

#### Caveat

I could not use ESBuild as the mochify bundler because it ran into a [dual package hazard](https://nodejs.org/api/packages.html#dual-package-hazard) issue with `sinon`.

- Browserify looks at the `browser` field in `package.json` and decides to load `./lib/sinon.js`.
- ESBuild checks the `exports` and chooses the `browser` field from there, which points to `./pkg/sinon-esm.js`.

I can override that behaviour with `conditions`, but then ESBuild complains that it cannot resolve `assert`, while Browserify will happily shim that dependency transparently.

Update: Consuming this with ESBuild doesn't work due to https://github.com/sinonjs/sinon/pull/2519. Reverting this change makes it work again. ¯\\_(ツ)_/¯ 

#### How to verify - mandatory

1. Check out this branch
2. `npm install`
3. `npm run lint`
4. `npm test`
5. `npm run test-cloud`

#### Checklist for author
